substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -343,395 +250,193 @@ private static void end(long addr) {
-
-    // (JDK 11)src/java.base/share/native/libzip/Deflater.c
-    // @formatter:off
-        // static jlong doDeflate(JNIEnv *env, jobject this, jlong addr,
-        //                        jbyte *input, jint inputLen,
-        //                        jbyte *output, jint outputLen,
-        //                        jint flush, jint params)
-        // @formatter:on
-    @SuppressWarnings({"cast"})
-    static long doDeflate(long addr,
-                    CCharPointer input, int inputLen,
-                    CCharPointer output, int outputLen,
-                    int flush, int params) {
+final class Util_java_util_zip_Deflater_JDK8OrEarlier {
 
-        z_stream strm = WordFactory.pointer(addr);
-        int inputUsed;
-        int outputUsed;
-        int finished = 0;
-        int setParams = params & 1;
-
-        strm.set_next_in(input);
-        strm.set_next_out(output);
-        strm.set_avail_in(inputLen);
-        strm.set_avail_out(outputLen);
-
-        if (setParams != 0) {
-            int strategy = (params >> 1) & 3;
-            int level = params >> 3;
-            int res = ZLib.deflateParams(strm, level, strategy);
-            if (res != ZLib.Z_OK() && res != ZLib.Z_BUF_ERROR()) {
-                throw new InternalError(CTypeConversion.toJavaString(strm.msg()));
-            }
-            if (res == ZLib.Z_OK()) {
-                setParams = 0;
-            }
-        } else {
-            int res = ZLib.deflate(strm, flush);
-            if (res != ZLib.Z_OK() && res != ZLib.Z_STREAM_END() && res != ZLib.Z_BUF_ERROR()) {
-                throw new InternalError(CTypeConversion.toJavaString(strm.msg()));
-            }
-            if (res == ZLib.Z_STREAM_END()) {
-                finished = 1;
-            }
-        }
-        inputUsed = inputLen - strm.avail_in();
-        outputUsed = outputLen - strm.avail_out();
-        return ((long) inputUsed) | (((long) outputUsed) << 31) | (((long) finished) << 62) | (((long) setParams) << 63);
+    static int update(Object obj, int len, z_stream strm) {
+        Target_java_util_zip_Deflater instance = SubstrateUtil.cast(obj, Target_java_util_zip_Deflater.class);
+        instance.off += instance.len - strm.avail_in();
+        instance.len = strm.avail_in();
+        return len - strm.avail_out();
     }
 }
 
-final class Util_java_util_zip_Deflater_JDK8OrEarlier {
-
+final class Util_java_util_zip_Inflater_JDK8OrEarlier {
     static int update(Object obj, int len, z_stream strm) {
-        Target_java_util_zip_Deflater instance = SubstrateUtil.cast(obj, Target_java_util_zip_Deflater.class);
+        JavaUtilZipSubstitutions.Target_java_util_zip_Inflater instance = SubstrateUtil.cast(obj, JavaUtilZipSubstitutions.Target_java_util_zip_Inflater.class);
         instance.off += instance.len - strm.avail_in();
         instance.len = strm.avail_in();
         return len - strm.avail_out();
     }
+
 }
 
+/** Dummy class to have a class with the file's name. */
```

Comment:
Please do not embed the substitutions into the JavaUtilZipSubstitutions dummy class.